### PR TITLE
A collection of small improvements

### PR DESCRIPTION
**Pull request by kellertuer**
While writing some answers in #264 I noticed that

* `get_stoping_criterion` might be nice
* `indicated_convergence` was not documented
* `get_state` would be nicer when it also had the recursive parameter (like `get_objective`)
* `get_state` replaces a few access functions

Internally

* maybe it would be reasonable to switch the state also to a `AbstractDecoratedManoptState` pattern like the objective, since there seems not much sense to keep the dispatch-with-THTT approach, since with an abstract decorator type it would be nicer to write _and_ unified with the objective. -> this is a bit more work for later
* one could also implement a more. precise way for convergence than just the string returned by `get_reason`. – maybe after we have MOI states internally (with their dependency) but not before.

I will collect a few small internal improvements here for a while.

### CODE REVIEW

File: src/plans/solver_state.jl [hunk 1]
```diff
@@ -121,16 +135,26 @@ function get_solver_return(o::ReturnManifoldObjective, s::AbstractManoptSolverSt
 end
 
 @doc raw"""
-    get_state(s::AbstractManoptSolverState)
+    get_state(s::AbstractManoptSolverState, recursive=truue)
```
Comment (mateuszbaran) [inline src/plans/solver_state.jl:138]:
```suggestion
    get_state(s::AbstractManoptSolverState, recursive::Bool=true)
```

File: src/plans/solver_state.jl [hunk 2]
```diff
@@ -121,16 +135,26 @@ function get_solver_return(o::ReturnManifoldObjective, s::AbstractManoptSolverSt
...
-return the undecorated [`AbstractManoptSolverState`](@ref) of the (possibly) decorated `s`.
-As long as your decorated state store the state within `s.state` and
-the [`dispatch_state_decorator`](@ref) is set to `Val{true}`,
-the internal state are extracted.
+If `recursive` is set to `false`, only the most outer decorator is taken away instead of all.
 """
-get_state(s::AbstractManoptSolverState) = _get_state(s, dispatch_state_decorator(s))
-_get_state(s::AbstractManoptSolverState, ::Val{false}) = s
-_get_state(s::AbstractManoptSolverState, ::Val{true}) = get_state(s.state)
+function get_state(s::AbstractManoptSolverState, recursive=true)
```
Comment (mateuszbaran) [inline src/plans/solver_state.jl:151]:
```suggestion
function get_state(s::AbstractManoptSolverState, recursive::Bool=true)
```

File: src/plans/solver_state.jl [hunk 3]
```diff
@@ -56,6 +56,20 @@ _get_message(s::AbstractManoptSolverState, ::Val{true}) = get_message(s.state)
 #INtroduce a default that there is no message
 _get_message(s::AbstractManoptSolverState, ::Val{false}) = ""
 
+@doc raw"""
+    get_stopping_criterion(ams::AbstractManoptSolverState)
+
+return the [`StoppingCriterion`](@ref) stored within the [`AbstractManoptSolverState`](@ref) `ams`.
```
Comment (mateuszbaran) [inline src/plans/solver_state.jl:62]:
```suggestion
Return the [`StoppingCriterion`](@ref) stored within the [`AbstractManoptSolverState`](@ref) `ams`.
```

File: docs/src/index.md [hunk 4]
```diff
@@ -8,13 +8,13 @@ CurrentModule = Manopt
...
 
-For a function $f:\mathcal M → ℝ$ defined on a [Riemannian manifold](https://en.wikipedia.org/wiki/Riemannian_manifold) $\mathcal M$ we aim to solve
+For a function ``f:\mathcal M → ℝ`` defined on a [Riemannian manifold](https://en.wikipedia.org/wiki/Riemannian_manifold) ``\mathcal M`` we aim to solve
 
 ```math
 \operatorname*{argmin}_{p ∈ \mathcal M} f(p),
 ```
 
-or in other words: find the point $x$ on the manifold, where $f$ reaches its minimal function value.
+or in other words: find the point ``x`` on the manifold, where ``f`` reaches its minimal function value.
```
Comment (mateuszbaran) [inline docs/src/index.md:17]:
```suggestion
or in other words: find the point ``p`` on the manifold, where ``f`` reaches its minimal function value.
```

**Comment by kellertuer** [comment]
Hm, I am a bit confused here. I could need this PR a lot on ManoptExamples just now but it seems my most recent commit made that every first run of the CI fails but every second one passes? Strange.

**Pull request merged**